### PR TITLE
[FAU-422] feat: add `fau/v1/degree-program/index` endpoint

### PR DESCRIPTION
**Pull request by shvlv**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature.

**What is the current behavior?** (You can also link to an open issue here)
https://inpsyde.atlassian.net/browse/FAU-442

**What is the new behavior (if this is a feature change)?**
New `index` endpoint.

image

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

**Other information**:

### CODE REVIEW

**Review by tyrann0us — APPROVED** [review]
Thanks! Beautiful! 💚

File: src/Infrastructure/RestApi/TranslatedDegreeProgramController.php [hunk 1]
```diff
@@ -176,6 +183,38 @@ public function get_items($request): WP_Error|WP_REST_Response
...
     }
 
+    public function getIndex(WP_REST_Request $request): WP_Error|WP_REST_Response
+    {
+        $criteria = CollectionCriteria::new()
+            ->withoutPagination();
+
+        $views = $this->degreeProgramCollectionRepository->findTranslatedCollection(
+            $criteria,
+            MultilingualString::DE
```
Comment (amiut) [inline src/Infrastructure/RestApi/TranslatedDegreeProgramController.php]: Although this is not defined by the client, does it make sense to have support for a `lang` query param. I guess we already have it here https://github.com/RRZE-Webteam/FAU-Studium-Common/blob/381097c700ebd3c42f4761dd215c8a9f2b641f4f/src/Infrastructure/RestApi/TranslatedDegreeProgramController.php#L122

**Review by zhyian — APPROVED** [review]
Thanks!

**Review by amiut — APPROVED** [review]
Nicely done, thanks a lot

**Pull request merged**